nengo_gui/page.py

```diff
@@ -50,6 +50,7 @@ def __init__(self, gui, filename, reset_cfg=False):
         self.finished = False  # should this Page be shut down
         self._sim = None       # the current nengo.Simulator
         self.rebuild = False   # should the model be rebuilt
+        self.sims_to_close = []  # list of sims that should be closed
```

Comment:
Perhaps overkill, but maybe this should be a set? So that if (for some odd reason) a simulator is added twice, it doesn't get closed twice. I think right now closing twice won't error, but that might not be true for all backends.